src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -2442,5 +2445,189 @@ private void InterpretLoadElement(int token)
+                        localVariableSet.SetVar<int>(i, stackItem.AsInt32());
+                        break;
+                    case TypeFlags.Int64:
+                    case TypeFlags.UInt64:
+                        localVariableSet.SetVar<long>(i, stackItem.AsInt64());
+                        break;
+                    case TypeFlags.IntPtr:
+                    case TypeFlags.UIntPtr:
+                        localVariableSet.SetVar<IntPtr>(i, stackItem.AsNativeInt());
+                        break;
+                    case TypeFlags.Single:
+                        localVariableSet.SetVar<float>(i, (float)stackItem.AsDouble());
+                        break;
+                    case TypeFlags.Double:
+                        localVariableSet.SetVar<double>(i, stackItem.AsDouble());
+                        break;
+                    case TypeFlags.ValueType:
+                    case TypeFlags.Nullable:
+                        throw new NotImplementedException();
+                    case TypeFlags.Enum:
+                        argumentType = argumentType.UnderlyingType;
+                        goto setvar;
+                    case TypeFlags.Class:
+                    case TypeFlags.Interface:
+                    case TypeFlags.Array:
+                    case TypeFlags.SzArray:
+                        localVariableSet.SetVar<object>(i, stackItem.AsObjectRef());
+                        break;
+                    default:
+                        // TODO: Support more complex return types
+                        throw new NotImplementedException();
+                }
+            }
+
+            if (callInfo.MethodAddress != IntPtr.Zero && callInfo.MethodAddressType == TypeLoaderEnvironment.MethodAddressType.Exact)
+            {
+                DynamicCallSignature dynamicCallSignature = new DynamicCallSignature(callInfo.CallingConvention, callInfo.LocalVariableTypes, callInfo.LocalVariableTypes.Length);
+                CallInterceptor.CallInterceptor.MakeDynamicCall(callInfo.MethodAddress, dynamicCallSignature, localVariableSet);
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+
+            TypeDesc returnType = callInfo.Signature.ReturnType;
+
+getvar:
+            switch (returnType.Category)
+            {
+                case TypeFlags.Void:
+                    // Do nothing!
+                    break;
+                case TypeFlags.Boolean:
+                    callInfo.ReturnValue = StackItem.FromInt32(localVariableSet.GetVar<int>(0));
+                    break;
+                case TypeFlags.Char:
+                    callInfo.ReturnValue = StackItem.FromInt32(localVariableSet.GetVar<char>(0));
+                    break;
+                case TypeFlags.SByte:
+                case TypeFlags.Byte:
+                    callInfo.ReturnValue = StackItem.FromInt32(localVariableSet.GetVar<sbyte>(0));
```

Comment:
Will this do the widening correctly? If a method returns `sbyte` 0xFF, shouldn't we widen to 0xFFFFFFFF, but if it returns `byte`, we should use `0x000000FF`?